modules/signing_worker/manifests/base.pp

```diff
@@ -0,0 +1,173 @@
+        default => [],
+    }
+
+    $supported_behaviors = $::hostname ? {
+        /^(tb-)?mac-v3-signing\d+/ => ['mac_sign', 'mac_sign_and_pkg', 'mac_geckodriver', 'mac_notarize'],
+        /^dep-mac-v3-signing\d+/ => ['mac_sign', 'mac_sign_and_pkg'],
+        default => [],
+    }
+
+    # Load hash of all the template variables
+    $worker_config = lookup("signingworker_config.${role}", Hash, undef, undef)
+    $role_config = lookup("signingworker_roles.${role}", Hash, undef, undef)
+
+    # For cloning the widevine repository
+    $widevine_user = lookup('widevine_config.user')
+    $widevine_key = lookup('widevine_config.key')
+
+    file { $tmp_requirements:
+        content => template('signing_worker/requirements.txt.erb'),
+        owner   =>  $signing_worker::user,
+        group   =>  $signing_worker::group,
+    }
+
+    vcsrepo { $widevine_clone_dir:
+      ensure   => present,
+      provider => git,
+      source   => "https://${widevine_user}:${widevine_key}@github.com/mozilla-services/widevine",
+    }
+    ->file { 'Remove widevine directory':
+      ensure  => absent,
+      path    => $widevine_clone_dir,
+      recurse => true,
+      purge   => true,
+      force   => true,
+    }
+
+    contain packages::virtualenv_python3_s3
+    python::virtualenv { "signingworker_${signing_worker::user}" :
+        ensure          => present,
+        version         => '3',
+        requirements    => $tmp_requirements,
+        venv_dir        => $virtualenv_dir,
+        ensure_venv_dir => true,
+        owner           => $signing_worker::user,
+        group           => $signing_worker::group,
+        timeout         => 0,
+        path            => [ '/bin', '/usr/bin', '/usr/sbin', '/usr/local/bin', '/Library/Frameworks/Python.framework/Versions/3.7/bin'],
+    }
+
+    case $::hostname {
+        /^dep-mac-v3-signing\d+/: {
+            file { "${certs_dir}/widevine_dep.crt":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.widevine_dep_crt')),
+            }
+            file { "${certs_dir}/dep-signing.keychain":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.dep_signing_keychain')),
+            }
+        }
+        default: {
+            file { "${certs_dir}/widevine_prod.crt":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.widevine_prod_crt')),
+            }
+            file { "${certs_dir}/nightly-signing.keychain":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.nightly_signing_keychain')),
```

Comment:
IIUC we've removed these keychains from the vault, so should this code me removed/commented out ?